packages/server-wallet/src/__test-with-peers__/ensure-objectives.test.ts

```diff
@@ -44,7 +44,11 @@ describe('EnsureObjectives', () => {
     'can successfully create a channel with the latency options: %o',
     async options => {
       messageService.setLatencyOptions(options);
-      const wallet = await Wallet.create(peerEngines.a, messageService);
+      const wallet = await Wallet.create(peerEngines.a, messageService, {
+        numberOfAttempts: 50,
+        initialDelay: 25,
+        multiple: 1.5,
+      });
```

Comment:
Referring to
> Change 2: The EnsureObjective tests use more aggressive retry options to ensure the test eventually passes

This change + the increased timeout is kind of worrying. Doing some math, `1.5^20*25 ~ 83131 ms`. So, the 20th attempt would have a delay of 80s, growing exponentially.

What's the probability that the test should flicker _due to the network parameters used_?  What's the expected time for the test to succeed? We should be aiming for it to succeed _quickly_, as to not slow down the whole test suite by up to 2 minutes.

Reply:
> This change + the increased timeout is kind of worrying. Doing some math, `1.5^20*25 ~ 83131 ms`. So, the 20th attempt would have a delay of 80s, growing exponentially. Using 50 attempts is a non-starter.

We're bounded by the jest test timeout so a value of `50` ensures that it will keep retrying until the test timeout. That seems ok to me? We could change it to a large value so that's clear.

> What's the probability that the test should flicker due to the network parameters used? What's the expected time for the test to succeed? We should be aiming for it to succeed quickly, as to not slow down the whole test suite by up to 2 minutes.

Currently, the `with-peers` tests are not the [slowest](https://app.circleci.com/pipelines/github/statechannels/statechannels/12328/workflows/c52a7d94-661e-4dcc-9bc5-da31f8bac737) so this isn't slowing down CI times at least.

I think we could drop the timeout back to 1 minute which seems more reasonable.

Reply:
ah, good point, this does not happen in the `test` step ...

If dropping the timeout down to 1 minute works, we should do it. But, looking now at the test times,
```
 PASS  src/__test-with-peers__/create-and-close-channel/ledger-funding.test.ts (20.006 s)
 PASS  src/__test-with-peers__/sync-objectives.test.ts (6.586 s)
{"level":50,"time":1617115456981,"pid":1307,"hostname":"5c4373d451d5","dbName":"server_wallet_test","walletVersion":"","msg":"Missing bytecode"}
 PASS  src/__test-with-peers__/crash-tolerance/crash-tolerance.test.ts (5.952 s)
{"level":50,"time":1617115462958,"pid":1307,"hostname":"5c4373d451d5","dbName":"server_wallet_test","walletVersion":"","msg":"Missing bytecode"}
 PASS  src/__test-with-peers__/create-and-close-channel/direct-funding.test.ts (5.995 s)
{"level":50,"time":1617115469069,"pid":1307,"hostname":"5c4373d451d5","dbName":"server_wallet_test","walletVersion":"","msg":"Missing bytecode"}
 PASS  src/__test-with-peers__/create-and-close-channel/fake-funding.test.ts (6.112 s)
 PASS  src/__test-with-peers__/ensure-objectives.test.ts (50.381 s)
```

`ensure-objectives` is by far the slowest of these tests, and it's close to 60s. Can we get 50s down? What causes it to be so slow?

Maybe setting `multiple: 1` would help. An _exponentially increasing_ backoff is important in production, to prevent retries from overwhelming an already-overwhelmed server, for instance. For `ensure-objectives`, we're not worried about that: a constant delay allows us to test the thing we want, without unduly waiting around for nothing.

Reply:
Setting the `multiple` to `1` will probably help a lot, let me take a look at that.